x-pack/plugins/triggers_actions_ui/public/application/lib/rule_api/map_filters_to_kql.ts

```diff
@@ -71,7 +71,11 @@ export const mapFiltersToKql = ({
     }
   }
   if (tagsFilter && tagsFilter.length) {
-    filters.push(`alert.attributes.tags:(${tagsFilter.join(' or ')})`);
+    filters.push(
+      `alert.attributes.tags:(${tagsFilter
+        .map((tag) => tag.replace(/([\)\(\<\>\}\{\"\:\\])/gm, '\\$&'))
```

Comment:
I am not sure if it is relevant to the PR but you can use `escapeKuery` from `@kbn/es-query`

Reply:
I tried to use `escapeKuery` but it seems like that does not escape `{}`. Looking at the source code, it does seem to support that.
```
function escapeSpecialCharacters(str: string) {
  return str.replace(/[\\():<>"*]/g, '\\$&'); // $& means the whole matched string
}
```
Also I noticed this regex pattern is used in security solutions: `x-pack/plugins/security_solution/public/management/common/utils.ts`. But yes I do think it would be ideal if we can just use  the same regex across the code base. So I wouldn't mind just using `escapeKuery`